src/app/clusters/occupancy-sensor-server/occupancy-sensor-server.cpp

```diff
@@ -0,0 +1,202 @@
+  EMBER_AF_NULL_MANUFACTURER_CODE, //manufacturerCode
+  .data.reported = {
+    1, //minInterval
+    EMBER_AF_PLUGIN_OCCUPANCY_SENSOR_SERVER_DEFAULT_OCCUPANCY_MAX_REPORT_PERIOD_S, //maxInterval
+    1 //reportableChange
+  }
+};
+*/
+//------------------------------------------------------------------------------
+// Plugin consumed callback implementations
+
+//******************************************************************************
+// Plugin init function
+//******************************************************************************
+void emberAfOccupancySensingClusterServerInitCallback(chip::EndpointId endpoint)
+{
+    HalOccupancySensorType deviceType;
+
+    checkForReportingConfig();
+    deviceType = halOccupancyGetSensorType(endpoint);
+
+    emberAfWriteAttribute(endpoint, ZCL_OCCUPANCY_SENSING_CLUSTER_ID, ZCL_OCCUPANCY_SENSOR_TYPE_ATTRIBUTE_ID, CLUSTER_MASK_SERVER,
+                          (uint8_t *) &deviceType, ZCL_ENUM8_ATTRIBUTE_TYPE);
+
+    uint8_t deviceTypeBitmap = 0;
+    switch (deviceType)
+    {
+    case HAL_OCCUPANCY_SENSOR_TYPE_PIR:
+        deviceTypeBitmap = EMBER_AF_OCCUPANCY_SENSOR_TYPE_BITMAP_PIR;
+        break;
+
+    case HAL_OCCUPANCY_SENSOR_TYPE_ULTRASONIC:
+        deviceTypeBitmap = EMBER_AF_OCCUPANCY_SENSOR_TYPE_BITMAP_ULTRASONIC;
+        break;
+
+    case HAL_OCCUPANCY_SENSOR_TYPE_PIR_AND_ULTRASONIC:
+        deviceTypeBitmap = (EMBER_AF_OCCUPANCY_SENSOR_TYPE_BITMAP_PIR | EMBER_AF_OCCUPANCY_SENSOR_TYPE_BITMAP_ULTRASONIC);
+        break;
+
+    case HAL_OCCUPANCY_SENSOR_TYPE_PHYSICAL:
+        deviceTypeBitmap = EMBER_AF_OCCUPANCY_SENSOR_TYPE_BITMAP_PHYSICAL_CONTACT;
+        break;
+
+    default:
+        break;
+    }
+    emberAfWriteAttribute(endpoint, ZCL_OCCUPANCY_SENSING_CLUSTER_ID, ZCL_OCCUPANCY_SENSOR_TYPE_BITMAP_ATTRIBUTE_ID,
+                          CLUSTER_MASK_SERVER, (uint8_t *) &deviceTypeBitmap, ZCL_BITMAP8_ATTRIBUTE_TYPE);
+
+    emberAfPluginOccupancyClusterServerPostInitCallback(endpoint);
+}
+
+//******************************************************************************
+// Notification callback from the HAL plugin
+//******************************************************************************
+void halOccupancyStateChangedCallback(EndpointId endpoint, HalOccupancyState occupancyState)
+{
+    if (occupancyState == HAL_OCCUPANCY_STATE_OCCUPIED)
+    {
+        emberAfOccupancySensingClusterPrintln("Occupancy detected");
+    }
+    else
+    {
+        emberAfOccupancySensingClusterPrintln("Occupancy no longer detected");
+    }
+
+    emberAfWriteAttribute(endpoint, ZCL_OCCUPANCY_SENSING_CLUSTER_ID, ZCL_OCCUPANCY_ATTRIBUTE_ID, CLUSTER_MASK_SERVER,
+                          (uint8_t *) &occupancyState, ZCL_BITMAP8_ATTRIBUTE_TYPE);
+    // emberAfPluginOccupancySensorServerOccupancyStateChangedCallback(
+    //   endpoint, occupancyState);
+}
+
+void emberAfPluginOccupancySensorServerStackStatusCallback(EndpointId endpoint, EmberStatus status)
```

Comment:
This seems like it should go away too, since it's a no-op.

Reply:
This is similar to how it is in https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/level-control/level-control.cpp#L1037. Probably those functions should be `__attribute__((weak))` so that they could be overriden by API users.